packages/native_pdf_renderer/android/src/main/kotlin/io/scer/pdf/renderer/NativePdfRendererPlugin.kt

```diff
@@ -166,9 +166,19 @@ class NativePdfRendererPlugin(private val registrar: Registrar) : MethodCallHand
                 val cropH = if (crop) call.argument<Int>("crop_height")!! else 0;
                 val cropW = if (crop) call.argument<Int>("crop_width")!! else 0;
 
+                val quality = call.argument<Int>("quality") ?: 100
+
                 val page = pages.get(pageId)
 
-                val results = page.render(width, height, color, format, crop, cropX, cropY, cropW, cropH).toMap
+                val tempOutFileExtension = when (format) {
+                    0 -> "jpg"
+                    1 -> "png"
+                    2 -> "webp"
+                    else -> "jpg"
+                }
+                val tempOutFile = File(registrar.context().cacheDir, "$randomFilename.$tempOutFileExtension")

```

Comment:
it is desirable to put the cache in an additional folder like a registrar.context().cacheDir + '/native_pdf_renderer_cache'

Reply:
Sorry, forgot about this PR.

I've updated the code to put the temporary files into a subdirectory named `native_pdf_renderer_cache`.

Also, I fixed the failing test, so now it should hopefully pass.